### PR TITLE
Added resource pipe and service for libraries

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**
Bulider PR: https://github.com/blackbaud/skyux-builder/pull/486
Builder Plugin PR: https://github.com/blackbaud/skyux-builder-plugin-skyux/pull/2
SKY UX Lookup PR (implementation demo): https://github.com/blackbaud/skyux-lookup/pull/3

### CODE REVIEW

File: src/app/public/modules/i18n/lib-resources.service.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      .map((info) => this.getStringForLocale(info, name, args));
+  }
+
+  public getDefaultString(name: string, ...args: any[]): string {
+    return this.getStringForLocale({ locale: 'en_US' }, name, args);
+  }
+
+  private getStringForLocale(
+    info: SkyAppLocaleInfo,
+    name: string, ...args: any[]
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/i18n/lib-resources.service.ts]: Missing line break

File: src/app/public/modules/i18n/lib-resources.service.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    return this.localeProvider
+      .getLocaleInfo()
+      .map((info) => this.getStringForLocale(info, name, args));
+  }
+
+  public getDefaultString(name: string, ...args: any[]): string {
+    return this.getStringForLocale({ locale: 'en_US' }, name, args);
+  }
+
+  private getStringForLocale(
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/i18n/lib-resources.service.ts]: Remove `getDefaultString` and make this public.

File: src/app/public/modules/i18n/lib-resources.service.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import {
+  Inject,
+  Injectable
+} from '@angular/core';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import 'rxjs/add/observable/merge';
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/modules/i18n/lib-resources.service.ts]: Remove this.

File: src/app/public/modules/i18n/lib-resources-provider.spec.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// import {
```
Comment (Blackbaud-BobbyEarl) [inline src/app/public/modules/i18n/lib-resources-provider.spec.ts]: Is this file necessary as it's all commented out?

Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/i18n/lib-resources-provider.spec.ts]: Nope! Thanks

Comment (Blackbaud-SteveBrush) [reply src/app/public/modules/i18n/lib-resources-provider.spec.ts]: @Blackbaud-BobbyEarl File removed.
